cpp/include/Ice/ClientAuthenticationOptions.h

```diff
@@ -0,0 +1,414 @@
+         * returns false, the connection will be aborted with an Ice::SecurityException.
+         *
+         * @remarks The server certificate chain is validated using the context object passed to the callback. When this
+         * callback is set, it replaces the default validation callback and must perform all necessary validation
+         * steps. If trustedRootCertificates is set, the passed context object will use them as the trusted root
+         * certificates for validation. This setting can be modified by the application using
+         * [SecTrustSetAnchorCertificates](https://developer.apple.com/documentation/security/1396098-sectrustsetanchorcertificates?language=objc)
+         *
+         * Example of setting serverCertificateValidationCallback:
+         * ```cpp
+         * auto initData = Ice::InitializationData {
+         *   ...
+         *   .clientAuthenticationOptions = ClientAuthenticationOptions{
+         *     .serverCertificateValidationCallback =
+         *       [](SecTrustRef trust, const IceSSL::ConnectionInfoPtr& info)
+         *       {
+         *          ...
+         *          return SecTrustEvaluateWithError(trust, nullptr);
+         *       }
+         * });
+         * auto communicator = Ice::initialize(initData);
+         * ```
+         *
+         * @param context A CtxtHandle representing the security context associated with the current connection. This
+         * context contains security data relevant for validation, such as the client's certificate chain and cipher
+         * suite.
+         * @param info The IceSSL::ConnectionInfoPtr object that provides additional connection-related data which might
+         * be relevant for contextual certificate validation.
+         * @return true if the certificate chain is valid and the connection should proceed; false if the certificate
+         * chain is invalid and the connection should be aborted.
+         * @throws Ice::SecurityException if the certificate chain is invalid and the connection should be aborted.
+         *
+         * [See
+         * SecTrustEvaluateWithError](https://developer.apple.com/documentation/security/2980705-sectrustevaluatewitherror?language=objc)
+         */
+        std::function<bool(CtxtHandle context, const IceSSL::ConnectionInfoPtr& info)>
+            serverCertificateValidationCallback;
+#elif defined(__APPLE__)
+        /**
+         * A callback that allows selecting the client's SSL certificate chain based on the target server host name.
+         *
+         * @remarks This callback is invoked by the SSL transport for each new outgoing connection before starting the
+         * SSL handshake to determine the appropriate client certificate chain. The callback should return a CFArrayRef
+         * that represents the client's certificate chain, or NULL if no certificate chain should be used for the
```

Comment:
nullptr?
```suggestion
         * that represents the client's certificate chain, or nullptr if no certificate chain should be used for the
```

Reply:
I have to check if `nullptr` works with CFArrayRef

Reply:
I thought that `nullptr` as a drop in replacement for `0` or `NULL`.